aws/usageReports/ebs/monthlyReport.go

```diff
@@ -0,0 +1,235 @@
+		} else {
+			l.Error("Query execution failed", map[string]interface{}{"error": err.Error()})
+		}
+		return nil, errors.GetErrorMessage(ctx, err)
+	}
+	return res, nil
+}
+
+// getSnapshotInfoFromEs gets information about an snapshot from previous report to put it in the new report
+func getSnapshotInfoFromES(ctx context.Context, snapshot utils.CostPerResource, account string, userId int) Snapshot {
+	var docType SnapshotReport
+	var snap = Snapshot{
+		SnapshotBase: SnapshotBase{
+			Id:          snapshot.Resource,
+			Description: "N/A",
+			State:       "N/A",
+			Encrypted:   false,
+			StartTime:   time.Time{},
+		},
+		Tags: make([]utils.Tag, 0),
+		Cost: 0,
+		Volume: Volume{
+			Id:   "N/A",
+			Size: -1,
+		},
+	}
+	res, err := getElasticSearchEbsSnapshot(ctx, account, snapshot.Resource,
+		es.Client, es.IndexNameForUserId(userId, IndexPrefixEBSReport))
+	if err == nil && res.Hits.TotalHits > 0 && len(res.Hits.Hits) > 0 {
+		err = json.Unmarshal(*res.Hits.Hits[0].Source, &docType)
+		if err == nil {
+			snap.Id = docType.Snapshot.Id
+			snap.Description = docType.Snapshot.Description
+			snap.State = docType.Snapshot.State
+			snap.Encrypted = docType.Snapshot.Encrypted
+			snap.StartTime = docType.Snapshot.StartTime
+			snap.Tags = docType.Snapshot.Tags
+		}
+	}
+	return snap
+}
+
+// fetchMonthlySnapshotsList sends in snapshotInfoChan the snapshots fetched from DescribeSnapshots
+// and filled by DescribeSnapshots and getSnapshotStats.
+func fetchMonthlySnapshotsList(ctx context.Context, creds *credentials.Credentials, snap utils.CostPerResource,
+	account, region string, snapshotChan chan Snapshot, userId int) error {
+	defer close(snapshotChan)
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	sess := session.Must(session.NewSession(&aws.Config{
+		Credentials: creds,
+		Region:      aws.String(region),
+	}))
+	svc := ec2.New(sess)
+	desc := ec2.DescribeSnapshotsInput{SnapshotIds: []*string{aws.String(getResourceId(snap))}}
+	snapshots, err := svc.DescribeSnapshots(&desc)
+	if err != nil {
+		snapshotChan <- getSnapshotInfoFromES(ctx, snap, account, userId)
+		return err
+	}
+	for _, snapshot := range snapshots.Snapshots {
+		snapshotChan <- Snapshot{
+			SnapshotBase: SnapshotBase{
+				Id:          aws.StringValue(snapshot.SnapshotId),
+				Description: aws.StringValue(snapshot.Description),
+				State:       aws.StringValue(snapshot.State),
+				Encrypted:   aws.BoolValue(snapshot.Encrypted),
+				StartTime:   aws.TimeValue(snapshot.StartTime),
+				Region:      region,
+			},
+			Tags: getSnapshotTag(snapshot.Tags),
+			Cost: snap.Cost,
+			Volume: Volume{
+				Id:   aws.StringValue(snapshot.VolumeId),
+				Size: aws.Int64Value(snapshot.VolumeSize),
+			},
+		}
+	}
+	return nil
+}
+
+// getEbsMetrics gets credentials, accounts and region to fetch EBS snapshots stats
+func fetchMonthlySnapshotsStats(ctx context.Context, snapshots []utils.CostPerResource, aa taws.AwsAccount, startDate time.Time) ([]SnapshotReport, error) {
```

Comment:
Function name is not the same as comment name

Reply:
fixed